modules/ags/src/main/scala/lucuma/ags/AgsAnalysis.scala

```diff
@@ -110,69 +121,79 @@ object AgsAnalysis {
       }
       val p              = if (withProbe) s"${guideProbe} " else ""
       val gs             = s"Guide Speed: ${guideSpeed.toString()}"
-      s"$qualityMessage$p$gs. vignetting: ${vignetting.head._2.toMicroarcsecondsSquared} µas^2"
-      s"$p $quality $gs. vignetting: ${vignetting.head._2.toMicroarcsecondsSquared} µas^2"
+      s"$qualityMessage$p$gs. vignetting: ${vignetting.toMicroarcsecondsSquared} µas^2"
+      s"$p $quality $gs. vignetting: ${vignetting.toMicroarcsecondsSquared} µas^2"
     }
-
-    def sortedVignetting: Usable = copy(vignetting = vignetting.sortBy(_._2))
   }
 
   object Usable {
-    val rankingOrder: Order[Usable] =
-      Order.by(u =>
-        (u.guideSpeed,
-         u.quality,
-         u.vignetting.minimumBy(_._2)._2.toMicroarcsecondsSquared,
-         u.target.gBrightness,
-         u.target.id
+    val rankOrder: Order[Usable] =
+      Order
+        .by(u =>
+          (u.guideSpeed,
+           u.quality,
+           u.vignetting.toMicroarcsecondsSquared,
+           u.target.gBrightness,
+           u.target.id
+          )
         )
-      )
 
+    val rankOrdering: Ordering[Usable] = rankOrder.toOrdering
   }
 
-  val rankingOrder: Order[AgsAnalysis] =
-    Order.from {
-      case (a: Usable, b: Usable) => Usable.rankingOrder.compare(a, b)
-      case (_: Usable, _)         => Int.MinValue
-      case (_, _: Usable)         => Int.MaxValue
-      case _                      => Int.MinValue
-    }
-
-  val rankingOrdering: Ordering[AgsAnalysis] = rankingOrder.toOrdering
-
-  extension (analysis: AgsAnalysis)
-    def posAngle: Option[Angle] = analysis match
-      case AgsAnalysis.Usable(_, _, _, _, v) => Some(v.head._1)
-      case _                                 => None
-
-  extension (analysis: Option[AgsAnalysis])
-    def posAngle: Option[Angle] = analysis.flatMap(_.posAngle)
+  extension (analysis: Option[AgsAnalysis]) def posAngle: Option[Angle] = analysis.map(_.posAngle)
 
   extension (results: List[AgsAnalysis])
+    // format: off
     /**
      * This method will sort the analysis for quality and internally for positions that give the
      * lowest vignetting.
      *
-     * Non usable positions wii be discarded.
+     * The rules are:
+     *   1. Foreach target, collect all the analyses
+     *   2. Then group by position angle
+     *   3. If any analysis withing a posAngle group is unusable, discard the posAngle because it
+     *      means it is not usable at at least one offset.
+     *   4. Otherwise, keep the posAngle within the group with the highest sorting - this is the
+     *      worst case for the posAngle.
+     * 5. For the target, choose the lowest sorting of the worst cases from step 4.
```

Comment:
Thanks for the comment 👍

Reply:
No problem. I didn't want to try to figure out what it was doing next time. 😆